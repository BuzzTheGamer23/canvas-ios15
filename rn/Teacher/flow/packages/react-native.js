// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
