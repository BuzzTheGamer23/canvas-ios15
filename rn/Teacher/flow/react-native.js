//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
