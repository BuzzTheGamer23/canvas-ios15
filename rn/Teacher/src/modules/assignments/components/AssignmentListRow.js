/**
* @flow
*/

import React, { Component, Element } from 'react'
import {
  View,
  TouchableHighlight,
  StyleSheet,
} from 'react-native'

import i18n from 'format-message'
import { formattedDueDateWithStatus } from '../../../common/formatters'
import Icon from './AssignmentListRowIcon'
import AssignmentDates from '../../../common/AssignmentDates'
import { Text } from '../../../common/text'

type Props = {
  assignment: Assignment,
  tintColor: string,
  onPress: (Assignment) => void,
}

export default class AssignmentListRow extends Component<any, Props, any> {
  onPress = () => {
    const assignment = this.props.assignment
    this.props.onPress(assignment)
  }

  dueDate (assignment: Assignment): Element<View> {
    const dates = new AssignmentDates(assignment)

    if (dates.availabilityClosed()) {
      return <Text style={styles.dueAtTitle}>{i18n('Availability: Closed')}</Text>
    }

    if (dates.hasMultipleDueDates()) {
      return <Text style={styles.dueAtTitle}>{i18n('Multiple Due Dates')}</Text>
    }

    const formattedDate = formattedDueDateWithStatus(dates.bestDueAt())
    return <Text style={styles.dueAtTitle}>{formattedDate}</Text>
  }

  ungradedBubble (assignment: Assignment): Element<View> {
    if (!assignment.needs_grading_count) {
      return <View />
    }

    const text = i18n('{count} ungraded', { count: assignment.needs_grading_count }).toUpperCase()
    return (
      <Text style={styles.ungradedText}>{text}</Text>
    )
  }

  render (): Element<View> {
    const assignment = this.props.assignment
    return (
      <View style={styles.row} key={assignment.id}>
        <TouchableHighlight style={styles.touchableHighlight} onPress={this.onPress} testID={`assignment-${assignment.id}`}>
          <View style={styles.container}>
            {assignment.published ? <View style={styles.publishedIndicatorLine} /> : <View />}
            <Icon published={assignment.published} tintColor={this.props.tintColor} />
            <View style={styles.textContainer}>
              <Text
                style={styles.title}
                ellipsizeMode='tail'
                numberOfLines={2}>{assignment.name}</Text>
              {this.dueDate(assignment)}
              {this.ungradedBubble(assignment)}
            </View>
          </View>
        </TouchableHighlight>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  row: {
    flex: 1,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'lightgrey',
  },
  touchableHighlight: {
    flex: 1,
  },
  container: {
    flex: 1,
    padding: 12,
    paddingLeft: 3,
    backgroundColor: 'white',
    alignItems: 'flex-start',
    flexDirection: 'row',
  },
  textContainer: {
    flex: 1,
  },
  title: {
    flex: 1,
    fontWeight: '600',
    color: '#2D3B45',
  },
  dueAtTitle: {
    color: '#8B969E',
    fontSize: 14,
    marginTop: 2,
  },
  ungradedText: {
    flex: 0,
    alignSelf: 'flex-start',
    fontSize: 11,
    fontWeight: '600',
    color: '#008EE2',
    borderRadius: 9,
    borderColor: '#008EE2',
    borderWidth: 1,
    backgroundColor: 'white',
    paddingTop: 3,
    paddingBottom: 1,
    paddingLeft: 6,
    paddingRight: 6,
    marginTop: 4,
    overflow: 'hidden',
  },
  publishedIndicatorLine: {
    backgroundColor: '#00AC18',
    position: 'absolute',
    top: 4,
    bottom: 4,
    left: 0,
    width: 3,
  },
})
