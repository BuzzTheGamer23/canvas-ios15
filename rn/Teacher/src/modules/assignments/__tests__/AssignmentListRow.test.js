/* @flow */

import 'react-native'
import React from 'react'
import AssignmentListRow from '../components/AssignmentListRow'
import moment from 'moment'

const template = {
  ...require('../../../__templates__/assignments'),
}

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  let assignment = template.assignment({ due_at: null })
  assignment.needs_grading_count = 0
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment} tintColor='#fff' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with needs_grading_count', () => {
  let assignment = template.assignment({ due_at: null })
  assignment.needs_grading_count = 5
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment} tintColor='#fff' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with closed due dates', () => {
  const one = moment().subtract(1, 'day').format()
  const two = moment().subtract(2, 'day').format()
  const three = moment().subtract(3, 'day').format()
  const assignment = template.assignment({
    lock_at: null,
    all_dates: [template.assignmentDueDate({ lock_at: one }), template.assignmentDueDate({ lock_at: two }), template.assignmentDueDate({ lock_at: three })],
  })

  let tree = renderer.create(
    <AssignmentListRow assignment={assignment} tintColor='#fff' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with open due dates', () => {
  const one = moment().add(1, 'day').format()
  const two = moment().add(2, 'day').format()
  const three = moment().add(3, 'day').format()
  const assignment = template.assignment({
    lock_at: null,
    all_dates: [template.assignmentDueDate({ lock_at: one }), template.assignmentDueDate({ lock_at: two }), template.assignmentDueDate({ lock_at: three })],
  })

  let tree = renderer.create(
    <AssignmentListRow assignment={assignment} tintColor='#fff' />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly with selected props', () => {
  let assignment = template.assignment({ due_at: null })
  assignment.needs_grading_count = 0
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment} tintColor='#fff' underlayColor='#eee' selectedColor='#f00'/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly published icon', () => {
  let assignment = template.assignment({ published: true })
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly not published icon', () => {
  let assignment = template.assignment({ published: false })
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly assignment icon', () => {
  let assignment = template.assignment({ submission_types: ['on_paper'] })
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly quiz icon', () => {
  let assignment = template.assignment({ submission_types: ['online_quiz'] })
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders correctly discussion icon', () => {
  let assignment = template.assignment({ submission_types: ['discussion_topic'] })
  let tree = renderer.create(
    <AssignmentListRow assignment={assignment}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
