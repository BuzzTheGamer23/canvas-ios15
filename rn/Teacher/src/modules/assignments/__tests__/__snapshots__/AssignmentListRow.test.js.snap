// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "marginLeft": -10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "height": undefined,
            },
            undefined,
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
          ],
        ]
      }
      testID="assignment-123456789"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          <View
            accessibilityLabel="Not Published"
            style={
              Object {
                "alignItems": "center",
                "flex": 0,
                "height": 32,
                "width": 46,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "position": "absolute",
                  },
                  Object {
                    "tintColor": "#fff",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "left": 20,
                  "position": "absolute",
                  "top": 10,
                  "width": 16,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 12,
                    "tintColor": "#8B969E",
                    "width": 12,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            learn to write code
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 14,
                  "marginTop": 2,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            No due date
          </Text>
          <View />
        </View>
        <View
          style={
            Object {
              "flex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#c7c7cc",
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                  "height": 10,
                  "marginLeft": 7,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View />
</View>
`;

exports[`renders correctly with closed due dates 1`] = `
<View>
  <View
    style={
      Object {
        "marginLeft": -10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "height": undefined,
            },
            undefined,
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
          ],
        ]
      }
      testID="assignment-123456789"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          <View
            accessibilityLabel="Not Published"
            style={
              Object {
                "alignItems": "center",
                "flex": 0,
                "height": 32,
                "width": 46,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "position": "absolute",
                  },
                  Object {
                    "tintColor": "#fff",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "left": 20,
                  "position": "absolute",
                  "top": 10,
                  "width": 16,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 12,
                    "tintColor": "#8B969E",
                    "width": 12,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            learn to write code
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 14,
                  "marginTop": 2,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            Availability: Closed
          </Text>
          <View />
        </View>
        <View
          style={
            Object {
              "flex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#c7c7cc",
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                  "height": 10,
                  "marginLeft": 7,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View />
</View>
`;

exports[`renders correctly with needs_grading_count 1`] = `
<View>
  <View
    style={
      Object {
        "marginLeft": -10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "height": undefined,
            },
            undefined,
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
          ],
        ]
      }
      testID="assignment-123456789"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          <View
            accessibilityLabel="Not Published"
            style={
              Object {
                "alignItems": "center",
                "flex": 0,
                "height": 32,
                "width": 46,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "position": "absolute",
                  },
                  Object {
                    "tintColor": "#fff",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "left": 20,
                  "position": "absolute",
                  "top": 10,
                  "width": 16,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 12,
                    "tintColor": "#8B969E",
                    "width": 12,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            learn to write code
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 14,
                  "marginTop": 2,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            No due date
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "alignSelf": "flex-start",
                  "backgroundColor": "white",
                  "borderColor": "#008EE2",
                  "borderRadius": 9,
                  "borderWidth": 1,
                  "color": "#008EE2",
                  "flex": 0,
                  "fontSize": 11,
                  "fontWeight": "600",
                  "marginTop": 4,
                  "overflow": "hidden",
                  "paddingBottom": 1,
                  "paddingLeft": 6,
                  "paddingRight": 6,
                  "paddingTop": 3,
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            5 UNGRADED
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#c7c7cc",
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                  "height": 10,
                  "marginLeft": 7,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View />
</View>
`;

exports[`renders correctly with open due dates 1`] = `
<View>
  <View
    style={
      Object {
        "marginLeft": -10,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "height": undefined,
            },
            undefined,
            Object {
              "borderBottomColor": "#C7CDD1",
              "borderBottomWidth": 0.5,
            },
          ],
        ]
      }
      testID="assignment-123456789"
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-start",
            }
          }
        >
          <View
            accessibilityLabel="Not Published"
            style={
              Object {
                "alignItems": "center",
                "flex": 0,
                "height": 32,
                "width": 46,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "position": "absolute",
                  },
                  Object {
                    "tintColor": "#fff",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "left": 20,
                  "position": "absolute",
                  "top": 10,
                  "width": 16,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 12,
                    "tintColor": "#8B969E",
                    "width": 12,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={2}
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "fontWeight": "600",
                },
                Object {
                  "fontFamily": ".SFUIDisplay-semibold",
                },
              ]
            }
          >
            learn to write code
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontSize": 16,
                },
                Object {
                  "color": "#8B969E",
                  "fontSize": 14,
                  "marginTop": 2,
                },
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
              ]
            }
          >
            Multiple Due Dates
          </Text>
          <View />
        </View>
        <View
          style={
            Object {
              "flex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#c7c7cc",
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                  "height": 10,
                  "marginLeft": 7,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View />
</View>
`;
